app/helpers/votes_helper.rb

```diff
@@ -0,0 +1,16 @@
+module VotesHelper
+  def user_voted?
+    vote = current_user.votes.where(['course_slug = ?', @course.slug]).first
+    !vote.nil? ? true : false
```

Comment:
This method can now be simplified (and made slightly more performant) with:

```ruby
def user_voted?(user, course_slug)
  user.votes.exists? course_slug: course_slug
end
```

(also note, don't access the `current_user` and `@course` variables directly as otherwise the helper is coupled to them being present – makes things harder to test, and more error prone)

Reply:
Done.